packages/text-field/src/vaadin-input-field-shared-styles.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { registerStyles, css } from '@vaadin/vaadin-themable-mixin/register-styles.js';
+
+registerStyles(
+  '',
+  css`
+    :host {
+      display: inline-flex;
+      outline: none;
+    }
+
+    :host::before {
+      content: '\\2003';
+      width: 0;
+      display: inline-block;
+      /* Size and position this element on the same vertical position as the input-field element
+          to make vertical align for the host element work as expected */
+    }
+
+    :host([hidden]) {
+      display: none !important;
+    }
+
+    :host(:not([has-label])) [part='label'] {
+      display: none;
+    }
+
+    [part='container'] {
+      display: flex;
+      flex-direction: column;
+      min-width: 100%;
+      max-width: 100%;
+      width: var(--vaadin-text-field-default-width, 12em);
```

Comment:
Don't know this pattern, just double-checking that this is intended. It seems the container would always be 100%, but then again the visual test screenshots seem to use a default size.

Reply:
There is a dedicated issue for it: #606. Let's discuss this question there.